packages/logger/src/formats.ts

```diff
@@ -0,0 +1,130 @@
+import { TransformableInfo } from 'logform';
+import stringify from 'json-stringify-safe';
+import { EOL } from 'os';
+import { format } from 'winston';
+
+const errorParamPos = 0;
+
+const LEVEL_EMOJI: Record<string, string> = {
+  info: '🍺',
+  warn: '❗️',
+  error: '🔥',
+  default: '🤷‍♂️',
+};
+
+/**
+ * Returns emoji format based on level.
+ */
+const emojiLevelFormat = format(
+  (info: TransformableInfo): TransformableInfo => {
+    const { level } = info;
+    const emoji = LEVEL_EMOJI[level] || LEVEL_EMOJI.default;
+    return { ...info, level: `${emoji} ${level}` };
+  },
+);
+
+/**
+ * Extracts parameters from winston.
+ */
+const getWinstonParams = (info: TransformableInfo): any[] | undefined => {
+  // Winston adds all parameters to 'splat' property which is accessible only by Symbol[splat]
+  return info[Symbol.for('splat') as any];
+};
+
+/**
+ * Returns stringified parameters.
+ * Returns undefined if parameters are undefined.
+ * If `logFormat` is `json`, Error object is omitted since it'll be included as `message`.
+ * If `logFormat` is `simple`, `EOL` is printed at the start of each string.
+ */
+const stringifyParams = (params: any[] | undefined, logFormat: 'simple' | 'json'): string[] | undefined => {
+  if (params === undefined) {
+    return params;
+  }
+
+  switch (logFormat) {
+    case 'json':
+      // Remove Error object if it's present where it's expected to be & stringify
+      if (params[errorParamPos] instanceof Error) {
+        return [
+          ...params.slice(0, errorParamPos),
+          ...params.slice(errorParamPos + 1),
+        ].map(p => stringify(p));
+      }
+
+      return params.map(p => stringify(p));
```

Comment:
```suggestion
      return params.map(stringify);
```

Reply:
This is the signature of the function:
```typescript
function stringify(obj: any, serializer?: stringify.EntryProcessor | null, indent?: string | number | null, decycler?: stringify.EntryProcessor): string
```

>Argument of type 'typeof stringify' is not assignable to parameter of type '(value: any, index: number, array: any[]) => string'.
  Types of parameters 'serializer' and 'index' are incompatible.
    Type 'number' is not assignable to type 'EntryProcessor'.

And if I actually try to run it, it fails cause it tries to call `serializer` since it receives a number (the index).

Reply:
@knor-el-snor I updated it wrapping `stringify` in a separate function & passing that into `.map`

Reply:
Oooh I didn't see it had a second parameter there. Makes sense then